src/strategy/CanaryDeploymentHelper.ts

```diff
@@ -0,0 +1,224 @@
+}
+
+export function getBaselineResourceName(name: string) {
+    return name + BASELINE_SUFFIX;
+}
+
+export function getStableResourceName(name: string) {
+    return name + STABLE_SUFFIX;
+}
+
+function UnsetsClusterSpecficDetails(resource: any) {
+
+    if (resource == null) {
+        return;
+    }
+
+    // Unsets the cluster specific details in the object
+    if (!!resource) {
+        const metadata = resource.metadata;
+        const status = resource.status;
+
+        if (!!metadata) {
+            const newMetadata = {
+                'annotations': metadata.annotations,
+                'labels': metadata.labels,
+                'name': metadata.name
+            };
+
+            resource.metadata = newMetadata;
+        }
+
+        if (!!status) {
+            resource.status = {};
+        }
+    }
+}
+
+function getNewCanaryObject(inputObject: any, replicas: number, type: string): object {
+    const newObject = JSON.parse(JSON.stringify(inputObject));
+
+    // Updating name
+    if (type === CANARY_LABEL_VALUE) {
+        newObject.metadata.name = getCanaryResourceName(inputObject.metadata.name)
+    } else if (type === STABLE_LABEL_VALUE) {
+        newObject.metadata.name = getStableResourceName(inputObject.metadata.name)
+    } else {
+        newObject.metadata.name = getBaselineResourceName(inputObject.metadata.name);
+    }
+
+    // Adding labels and annotations.
+    addCanaryLabelsAndAnnotations(newObject, type);
+
+    // Updating no. of replicas
+    if (isSpecContainsReplicas(newObject.kind)) {
+        newObject.spec.replicas = replicas;
+    }
+
+    return newObject;
+}
+
+function isSpecContainsReplicas(kind: string) {
+    return !isEqual(kind, KubernetesWorkload.pod, StringComparer.OrdinalIgnoreCase) &&
+        !isEqual(kind, KubernetesWorkload.daemonSet, StringComparer.OrdinalIgnoreCase) &&
+        !helper.isServiceEntity(kind)
+}
+
+function addCanaryLabelsAndAnnotations(inputObject: any, type: string) {
+    const newLabels = new Map<string, string>();
+    newLabels[CANARY_VERSION_LABEL] = type;
+
+    helper.updateObjectLabels(inputObject, newLabels, false);
+    helper.updateSelectorLabels(inputObject, newLabels, false);
+
+    if (!helper.isServiceEntity(inputObject.kind)) {
+        helper.updateSpecLabels(inputObject, newLabels, false);
+    }
+}
+
+function createCanaryObjectsArgumentString(files: string[], includeServices: boolean) {
+    const kindList = new Set<string>();
+    const nameList = new Set<string>();
+
+    files.forEach((filePath: string) => {
+        const fileContents = fs.readFileSync(filePath);
+        yaml.safeLoadAll(fileContents, function (inputObject) {
+            const name = inputObject.metadata.name;
+            const kind = inputObject.kind;
+            if (helper.isDeploymentEntity(kind)
+                || (includeServices && helper.isServiceEntity(kind))) {
+                const canaryObjectName = getCanaryResourceName(name);

```

Comment:
Before adding canary and base line object, first we need to check resoruces exist if not don't add in the list.